### PR TITLE
CI: Bump actions version; fix Node 12 deprecation warning

### PR DESCRIPTION
**Pull request by kbdharun**
### Purpose

This PR fixes the Node 12 deprecation shown under the action runs of update-docs-translations.yaml by bumping the version of `actions/checkout` from `v2` (Node 12) to `v3` (Node 16). 

This PR also updates `actions/setup-go` to `v4`.

Closes #8832

### Testing

These changes haven't been tested yet, but they are non-breaking changes.

### CODE REVIEW

File: .github/workflows/build-syncthing.yaml [hunk 1]
```diff
@@ -57,9 +57,9 @@ jobs:
 
       - uses: actions/checkout@v3
 
-      - uses: actions/setup-go@v3
+      - uses: actions/setup-go@v4
         with:
-          go-version: ${{ matrix.go }}
```
[the remark this answers is not in the document]
Comment (kbdharun) [reply .github/workflows/build-syncthing.yaml:62]: Done, reverted it.

**Review by calmh — APPROVED** [review]

**Comment by kbdharun** [comment]
> I wasn't allowed to push changes to your PR branch, but if you merge with `main` there's one more setup-go that needs fixing now. Sorry about that.

Sure, I will try merging with `main` now.

**Comment by calmh** [comment]
Right, now if you peek at build-syncthing.yaml you'll see there's an additional actions/setup-go@v3 that could be fixes in this same PR :)

**Comment by kbdharun** [comment]
> Right, now if you peek at build-syncthing.yaml you'll see there's an additional actions/setup-go@v3 that could be fixes in this same PR :)

I will sync the branch with `main` again.

**Review by calmh — APPROVED** [review]

**Pull request merged**